src/lib/y2partitioner/dialogs/disk_clone.rb

```diff
@@ -0,0 +1,193 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+require "y2partitioner/dialogs/popup"
+require "y2partitioner/device_graphs"
+require "y2partitioner/confirm_recursive_delete"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Dialogs
+    # Dialog for cloning a device
+    class DiskClone < Popup
+      # @return [Actions::Controllers::DiskDevice]
+      attr_reader :controller
+
+      # Constructor
+      #
+      # @param controller [Actions::Controllers::DiskDevice]
+      def initialize(controller)
+        textdomain "storage"
+
+        @controller = controller
+      end
+
+      # @macro seeDialog
+      def title
+        # TRANSLATORS: %{name} is replaced by a disk device name (e.g., /dev/sda)
+        format(_("Clone partition layout of %{name}"), name: controller.device.name)
+      end
+
+      # @macro seeDialog
+      # @see DevicesSelector
+      def contents
+        @contents ||= VBox(DevicesSelector.new(controller))
+      end
+
+      # Widget to select devices for cloning
+      class DevicesSelector < CWM::MultiSelectionBox
+        include ConfirmRecursiveDelete
+
+        # Constructor
+        #
+        # @param controller [Actions::Controllers::DiskDevice]
+        def initialize(controller)
+          textdomain "storage"
+
+          @controller = controller
+          self.handle_all_events = true
+        end
+
+        # @return [String]
+        def label
+          _("Available target disks:")
+        end
+
+        # @return [Array<Integer, String>]
+        def items
+          controller.suitable_devices_for_cloning.map { |d| [d.sid, label_for(d)] }
+        end
+
+        # Checks whether any device was selected
+        #
+        # @note A confirmation popup is shown when deleting devices is needed
+        #   for cloning into the selected devices.
+        #
+        # @return [Boolean]
+        def validate
+          return confirm? if selected_devices?
+
+          Yast::Popup.Error(_("Select a target disk for creating a clone"))
+          false
+        end
+
+        # Saves the selected devices into the controller
+        def store
+          controller.selected_devices_for_cloning = selected_devices
+        end
+
+        # FIXME: The help handle does not work without a wizard
+        #
+        # This handle should belongs to the dialog
+        def handle(event)
+          return nil if event["ID"] != :help
+
+          Yast::Wizard.ShowHelp(help)
```

Comment:
I think this is bad idea, what if more widgets have help? it means that everyone shows his own help and only one can rule them all.